synapse/types.py

```diff
@@ -12,11 +12,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import string
 
 from synapse.api.errors import SynapseError
 
 from collections import namedtuple
+import re
```

Comment:
I normally have my editor configured to sort imports automatically, but I have turned it off for synapse so the diffs stay under control. Perhaps it would be worthwhile to run `isort` on the repo at some point.